src/deploy/functions/services/firestore.ts

```diff
@@ -0,0 +1,25 @@
+import * as backend from "../backend";
+import * as firestore from "../../../gcp/firestore";
+import { FirebaseError } from "../../../error";
+
+/**
+ * Sets a firestore event trigger's region to the firestore database region.
+ * @param endpoint the firestore endpoint
+ */
+export async function ensureFirestoreTriggerRegion(
+  endpoint: backend.Endpoint & backend.EventTriggered
+): Promise<void> {
+  const db = await firestore.getDatabase(
+    endpoint.project,
+    endpoint.eventTrigger.eventFilters?.database || "(default)"
+  );
```

Comment:
q: in `functions/services/database` we don't make similar guarantee like we've done here by making a call to get RTDB info. What's the difference here?

Reply:
For RTDB, we explicitly call it out in the (docs)[https://firebase.google.com/docs/functions/beta/database-events#setting_the_function_location]. It would be a little harder to have verification since the GET (Rest API)[https://firebase.google.com/docs/reference/rest/database/database-management/rest] requires the region to be in the query. We _could_ write some custom logic with a LIST request, but on the original proposal we said that docs should mitigate this risk of deployment failure.